### ISSUE TITLE
Nonce API is inflexible and inaccurately typed

### ISSUE THREAD

**Issue by mantariksh** [opengovsg/sgid-client#61]
tl;dr there are 2 problems:
1. No util function to generate nonce
2. Nonce returned from `authorizationUrl` is inconveniently typed as `string | undefined`

I want to implement the following flow for generating the authorisation URL:
1. Controller calls a service to generate required params for authorisation URL (code verifier/challenge, nonce)
3. Controller calls another service to generate authorisation URL with the params generated in step 1
4. Controller saves code verifier and nonce in a cookie

But in step 1, there is no API to generate a nonce separately from auth URL. Workarounds are either:
1. Call `.authorizationUrl` just to get a random nonce. But this returns a nonce of type `string | undefined`, which forces me to write unnecessary null checks like `return nonce ?? ''`
2. Call `generateCodeVerifier` because it uses the same underlying function to generate the nonce. But this requires knowledge of sgID client internals.

Suggestion: expose a separate util function to generate a nonce, as with code verifier and code challenge.
